tests/ml_models/test_sentencetransformermodel_pytest.py

```diff
@@ -658,5 +660,23 @@ def test_zip_model_with_license():
     clean_test_folder(TEST_FOLDER)
 
 
+@pytest.fixture
+def mock_opensearch_client():
```

Comment:
We might not want to mock Opensearch client. Tests we are running are integration tests. Hence, mocking might not be useful.

Reply:
Seems like I am wrong. In this case, we don't need the client.

Reply:
@Vinay-Vinod Thanks for raising the PR. Could you please make sure if this PR addresses all these:

```
What solution would you like?
I would like to EITHER 1/ only provide the model_id when I instantiate the SentenceTransformerModel and it gets used in all class methods where applicable, rather than randomly defaulting back to "distilbert-tas-b" OR 2/ use save_as_pt as a static method and never instantiate the class at all. Also it would be nice if I could specify the output location of make_model_config_json.
```

1. only provide the model_id when I instantiate the SentenceTransformerModel and it gets used in all class methods where applicable, rather than randomly defaulting back to "distilbert-tas-b"
2. In `save_as_pt` method, if no other model id is given as function parameter, it will use the model id which was initialized with SentenceTransformerModel class. If still a model is given as a parameter then in  `save_as_pt` method we'll use that model id. Please let me know if you have any confusion.
3. it would be nice if I could specify the output location of make_model_config_json  --> Can we take care of this one too? Or are you planning to raise another PR for this?